### PR TITLE
fuzzing: fix tests

### PR DESCRIPTION
**Pull request by giuseppe**
adjust fuzzing tests to the new API.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
@alexlarsson PTAL

File: fuzzing/main.c [hunk 1]
```diff
@@ -26,97 +26,44 @@ int LLVMFuzzerInitialize(int *argc, char ***argv)
...
-	if (IS_ERR(cfs_ino))
-		return true;
-
-
-	cfs_ino_data = lcfs_inode_data(ctx, cfs_ino, &cfs_ino_data_buf);
-	if (IS_ERR(cfs_ino_data))
-		return true;
-
-	payload = lcfs_dup_payload_path(ctx, cfs_ino);
+	payload = lcfs_dup_payload_path(test_ctx->ctx, s_ino, 0);
```
Comment (alexlarsson) [inline fuzzing/main.c:45]: You can't call both get_backing and dup_payload on the same inode. It needs to check if its a symlink or regular first, and same for get_dir().

File: fuzzing/main.c [hunk 2]
```diff
@@ -26,97 +26,44 @@ int LLVMFuzzerInitialize(int *argc, char ***argv)
...
-	if (test_ctx->recursion_left <= 0)
+	s_ino = lcfs_get_ino_index(test_ctx->ctx, ino, &buffer);
+	if (IS_ERR(s_ino))
 		return true;
 
-	cstr = name;
-	/* Consume the C string.  */
-	while (*cstr)
-		cstr++;
+	lcfs_get_backing(test_ctx->ctx, s_ino, 0, &out_size, &out_path);
```
Comment (alexlarsson) [inline fuzzing/main.c:43]: This leaks out_path.

File: fuzzing/main.c (hunk 1 above)
Comment (alexlarsson) [reply fuzzing/main.c:45]: Otoh, maybe the callees should verify that and return error in that case.

File: fuzzing/main.c (hunk 2 above)
Comment (giuseppe) [reply fuzzing/main.c]: thanks, fixed now

File: fuzzing/main.c (hunk 1 above)
Comment (giuseppe) [reply fuzzing/main.c:47]: yes, I think it is safer that the calles verify that and do not crash

**Pull request merged**